internal/internal_workflow.go

```diff
@@ -784,6 +801,10 @@ func (c *channelImpl) ReceiveAsyncWithMoreFlag(valuePtr interface{}) (ok bool, m
 	}
 }
 
+func (c *channelImpl) Empty() bool {
+	return c.closed || (len(c.buffer) == 0 && len(c.blockedSends) == 0 && c.recValue == nil)
```

Comment:
In Go, technically a closed channel with items still in the buffer has a len > 0 and is not considered empty. Didn't check if `closed` field here represents when a user calls close or after the user has called closed _and_ all items are gone from the buffer.